### PR TITLE
Fix multiselect custom field in campaign condition

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5537
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

Bugfix/feature donate by www.newcom4u.ch

[//]: # ( Required: )
#### Description:

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a multi-liste custom field
2. Create contact and add all values from multiselect
3. Create campagin with lead field condition and multiselect 
4. Action after condition was not trigger

#### Steps to test this PR:
1.  Repeat all steps
2.  See If camapign condition work properly

### CODE REVIEW

File: app/bundles/LeadBundle/Entity/LeadFieldRepository.php [hunk 1]
```diff
@@ -192,6 +192,23 @@ public function compareValue($lead, $field, $value, $operatorExpr)
...
+                );
+
+                $value = trim($value, "'");
+                if (substr($operatorExpr, 0, 3) === 'not') {
+                    $operator = 'NOT REGEXP';
+                } else {
+                    $operator = 'REGEXP';
+                }
+                $expr->add(
+                    'l.'.$field." $operator '\\\\|?$value\\\\|?'"
```
Comment (alanhartless) [inline app/bundles/LeadBundle/Entity/LeadFieldRepository.php]: This will break if $value has any of MySQL's regex special characters.

Comment (kuzmany) [reply app/bundles/LeadBundle/Entity/LeadFieldRepository.php]: Added the same escape cleaner like https://github.com/mautic/mautic/blob/ad1e4481d0db3374e14172c64736196c8e7b76ef/app/bundles/LeadBundle/Entity/LeadListRepository.php#L1681

**Pull request merged**